token/cli/src/bench.rs

```diff
@@ -484,16 +484,9 @@ async fn send_messages(
         return Ok(());
     }
 
-    let (_blockhash, fee_calculator, _last_valid_block_height) = config
-        .rpc_client
-        .get_recent_blockhash_with_commitment(config.rpc_client.commitment())
-        .await?
-        .value;
-
-    lamports_required += messages
-        .iter()
-        .map(|message| fee_calculator.calculate_fee(message))
-        .sum::<u64>();
+    for message in messages.iter() {
+        lamports_required += config.rpc_client.get_fee_for_message(message).await?;
```

Comment:
This will probably slow things down unfortunately.  Check out the recent discussion in new-hire-learning-corner on Discord.  It looks like this is always called with the same kind of message, so you can fetch the fee for the first message, and then assume that all others will cost the same amount.

Also, this will probably weirdly error because you need a blockhash provided on the message in order to fetch a fee. I've been had on that *a lot*, so be sure to fetch a blockhash, get one fee, and multiply to greatness.

What do you think?

Reply:
after reviewing what the code is doing in more detail, yep this all makes sense to me